R/admin-panel.R

```diff
@@ -487,20 +487,42 @@ zip_dir <- function(dir, output_file) {
 df_all_results <- function(results_dir) {
   files <- list_results_files(results_dir, full.names = TRUE)
   if (length(files) == 0L) return(data.frame())
-  data <- lapply(files, readRDS)
-  data_df <- lapply(data, as.data.frame)
-  any_cols_duplicated <- any(vapply(data_df,
-                                    function(df) anyDuplicated(names(df)),
-                                    integer(1)) > 0L)
-  if (any_cols_duplicated) {
-    msg <- "CSV export cannot cope with duplicated fields in results objects."
-    shiny::showNotification(msg, type = "error")
-    stop(msg)
-  }
-  df <- do.call(plyr::rbind.fill, data_df)
-  df <- df[order(df$session.current_time, decreasing = FALSE), ]
-  df
+  df <- purrr::map_dfr(files, function(results){
```

Comment:
It looks conceptually good and I'm happy if you say that the change maintains back-compatibility. However, I am seeing that one of the testthat tests is failing, suggesting that perhaps back-compatibility might be broken in some way... Have you checked it?